### PR TITLE
Various chatlog improvements

### PR DESCRIPTION
**Pull request by pali6**
## About the PR 
This PR adds, readds, improves and fixes several things about the chatlog thing in the bottom right corner of your screen:

- Fixed darkmode submenus not being dark
- Increased number of hightlightable terms from 5 to 10
- Added a toggle for changing the emoji style between default (Windows emoji) and Twemoji (used on Twitter and Discord)
- Added a toggle for disabling erasing of old messages (currently if you get 4000 messages the first half gets Thanos-snapped, this might be undesirable for those who wish to keep logs which brings us to...)
- Fixed up, improved and re-enabled the Save Log button in the chatlog

How the menu looks after this PR:
image

Emoji style comparison:
dreamseeker_CLsvVapnFH

## Why's this needed? 
The save log functionality was removed due to it crashing people. This should hopefully be fixed now and it should also work better overall.
The message limit toggle is useful for people who wish to keep long logs of their rounds.
5 highlight terms often doesn't seem like enough, I don't think there's any harm in increasing them to 10.
The emoji thing is probably mostly useless, I admit. If people feel like the button would just clutter up the menu I'll remove it.

## Changelog

```changelog
(u)pali
(*)You can now save game logs in the chatlog menu (the cogwheel in the top right corner of the chatlog).<br>You can also enter 5 more words for highlighting, toggle emoji style and toggle deletion of old messages.
```

### CODE REVIEW

File: browserassets/js/browserOutput.js [hunk 1]
```diff
@@ -913,25 +943,20 @@ $(function() {
         } else {
             xmlHttp = new ActiveXObject("Microsoft.XMLHTTP");
         }
-        xmlHttp.open('GET', 'browserOutput.css', false);
+        xmlHttp.open('GET', 'http://cdn.goonhub.com/css/browserOutput.css', false);
```
Comment (pali6) [inline browserassets/js/browserOutput.js:946]: It is intentional that this has the cdn url here. Using the relative url keeps freezing your client indefinitely, at least locally.

**Comment by ZeWaka** [comment]
How do ZWJ windows-only sequences fair? such as 🐱‍👤 and 🐱‍💻 - probably just show the components?

**Comment by pali6** [comment]
They show up properly when using the font and as parts when using Twemoji:
image
Considering that they seem to be things that Microsoft just made up themselves it doesn't surprise me that Twemoji maintainers decided not to support them:
https://github.com/twitter/twemoji/issues/373

**Comment by pali6** [comment]
Tomorrow pali should fix mismatched indentation (our js files use spaces but our vscode setup doesn't acknowledge that).

**Pull request merged**